cryptoprotocol/src/main/java/de/upb/crypto/incentive/cryptoprotocol/model/PublicParameters.java

```diff
@@ -0,0 +1,55 @@
+package de.upb.crypto.incentive.cryptoprotocol.model;
+
+
+import de.upb.crypto.craco.prf.aes.AesPseudorandomFunction;
+import de.upb.crypto.craco.sig.interfaces.SignatureKeyPair;
+import de.upb.crypto.craco.sig.sps.eq.SPSEQSignatureScheme;
+import de.upb.crypto.craco.sig.sps.eq.SPSEQSigningKey;
+import de.upb.crypto.craco.sig.sps.eq.SPSEQVerificationKey;
+import de.upb.crypto.math.interfaces.structures.GroupElement;
+import de.upb.crypto.math.pairings.generic.BilinearGroup;
+
+/**
+ * a class representing the public parameters of the 2020 incentive system
+ */
+
+public class PublicParameters
+{
+    private BilinearGroup bg;
+    private GroupElement w;
+    private GroupElement h7;
+    private AesPseudorandomFunction prf; // not in paper, but we need to store PRF that is used in incentive system instance somewhere
+    private SPSEQSignatureScheme spsEq; // same here for SPS-EQ scheme
+    private SignatureKeyPair<SPSEQVerificationKey, SPSEQSigningKey> spsEqKeyPair;
```

Comment:
Don't store secret keys in public parameters 😉. Should be generated during provider's key generation and stored in provider's public/secret key.

Reply:
I dont actually know how this line made it until here ...

Reply:
... but I do know why I wrote it initially.
Some past version of myself grow fond of the idea to generate the SPS-EQ key pair in the Setup.trustedSetup method.
Don't know why but I fixed it now.